var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,531 @@
 import sys
 
 
-class Sundials(Package):
+class Sundials(CMakePackage):
     """SUNDIALS (SUite of Nonlinear and DIfferential/ALgebraic equation
     Solvers)"""
 
-    homepage = "http://computation.llnl.gov/casc/sundials/"
-    url = "http://computation.llnl.gov/projects/sundials-suite-nonlinear-differential-algebraic-equation-solvers/download/sundials-2.6.2.tar.gz"
+    homepage = "https://computation.llnl.gov/projects/sundials"
+    url = "https://computation.llnl.gov/projects/sundials/download/sundials-2.7.0.tar.gz"
 
-    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3')
+    # ==========================================================================
+    # Versions
+    # ==========================================================================
+
+    version('3.0.0-beta-2',
+            git='https://github.com/LLNL/sundials.git', tag='v3.0.0-beta-2')
+    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3', preferred=True)
     version('2.6.2', '3deeb0ede9f514184c6bd83ecab77d95')
 
+    # ==========================================================================
+    # Variants
+    # ==========================================================================
+
+    # SUNDIALS solvers
+    variant('cvode',  default=True,
+            description='Enable CVODE')
+    variant('cvodes', default=True,
+            description='Enable CVODES')
+    variant('arkode', default=True,
+            description='Enable ARKode')
+    variant('ida',    default=True,
+            description='Enable IDA')
+    variant('idas',   default=True,
+            description='Enable IDAS')
+    variant('kinsol', default=True,
+            description='Enable KINSOL')
+
+    # Real precision type
+    variant(
+        'precision',
+        default='double',
+        description='''real type precision''',
+        values=('single', 'double', 'extended'),
+        multi=False
+    )
+
+    # Index type
+    variant(
+        'indextype',
+        default='int64_t',
+        description='''index integer type''',
+        values=('int32_t', 'int64_t'),
+        multi=False
+    )
+
+    # Parallelism
     variant('mpi',     default=True,
-            description='Enable MPI parallelism')
-    variant('lapack',  default=True,
-            description='Use external BLAS/LAPACK libraries')
-    variant('klu',     default=False,
+            description='Enable MPI parallel vector')
+    variant('openmp',  default=False,
+            description='Enable OpenMP parallel vector')
+    variant('pthread', default=False,
+            description='Enable Pthreads parallel vector')
+    variant('cuda',    default=False,
+            description='Enable CUDA parallel vector')
+    variant('raja',    default=False,
+            description='Enable RAJA parallel vector')
+
+    # External libraries
+    variant('blas',       default=False,
```

Comment:
why do we need internal blas at all? I would remove the two variants all together...  Imaging what could happen if one uses `sundials` as a part of a bigger DAG (i.e. `dealii`) which uses `petsc`, `trilinos`, `mumps` etc where everyone use exactly the same Blas/Lapack, but now you would have different Blas/Lapack within `sundials` itself. So if you have dynamic linking, you can easily get into troubles because you don't have control over what blas/lapack is used within sundials (threaded / openmp or not).

Reply:
generally, 99% packages in Spack do not build any dependencies themselves, look at `petsc` which is able to compile all of its dependencies but we provide them via Spack for a good reason.

Reply:
another way to look at it is that **Spack promises that there is one and only one version/variant/configuration of each dependency within the DAG**. If you build internal Blas/Lapack, this is no longer the case...

Reply:
Sundials does not build an internal `blas` and only needs to link to a `blas` library when building when an external solver requires `blas` (e.g. `superlu_mt`). Activating the `blas` variant sets the path to whatever `blas` the user is providing.

Reply:
> an external solver requires blas (e.g. superlu_mt).

oh, then you don't need these variants at all. If `superlu_mt` is there and it needs blas/lapack, you will have it in DAG.

Reply:
@tgamblin : if some dependency in the DAG needs blas/lapack, `spec['blas'].libs` will still work, right, meaning there is no need to declare that the root node depends on blas/lapack?

Reply:
> Activating the blas variant sets the path to whatever blas the user is providing.

atlernative is still to remove those variants and add something like
```
depends_on('blas', when='+superlu_mt')
depends_on('lapack', when='+superlu_mt')
```

Reply:
It is also possible to build `superlu_mt` with its own internal `blas` so it doesn't necessarily depend on using `blas`. So it seems like we don't need the `blas` variant or the `depends_on` in the package.

Reply:
> It is also possible to build superlu_mt with its own internal blas

that's bad for the reasons outlined above... Luckily by default it will use blas from Spack. Maybe you can make sure it is this way here by writing
```
depends_on(superlu_mt+blas,...)
```

Reply:
`superlu_mt` encourages users to not use the internal `blas` unless they have to and since Spack will provides a `blas` installation your suggestion sounds like the way to go.